File: androidApp/src/main/java/com/mdgroup/nasawallpapers/core/di/ViewModelsModule.kt
package com.mdgroup.nasawallpapers.core.di

import android.content.res.Resources
import com.mdgroup.nasawallpapers.domain.interactors.LicenseInteractor
import com.mdgroup.nasawallpapers.domain.interactors.WallpaperInteractor
import com.mdgroup.nasawallpapers.presentation.viewmodels.BookmarksViewModel
import com.mdgroup.nasawallpapers.presentation.viewmodels.LicensesViewModel
import com.mdgroup.nasawallpapers.presentation.viewmodels.WallpaperViewModel
import com.mdgroup.nasawallpapers.presentation.viewmodels.WallpapersViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val viewModelsModule = module {

    viewModel { WallpapersViewModel(interactor = get<WallpaperInteractor>()) }

    viewModel { (date: String?) -> WallpaperViewModel(date = date, resources = get<Resources>(), interactor = get<WallpaperInteractor>()) }

    viewModel { BookmarksViewModel(interactor = get<WallpaperInteractor>()) }

    viewModel { LicensesViewModel(interactor = get<LicenseInteractor>()) }
}

File: iosApp/iosApp/core/di/ViewModelsModule.swift
//
//  ViewModelsModule.swift
//  iosApp
//
//  Created by Pavlo Kravchenko on 17.02.2023.
//  Copyright © 2023 mdgroup. All rights reserved.
//

import Foundation
import shared
import SwiftUI

func getWallpapersViewModel() -> WallpapersViewModel {
    
    let interactor = koin.get(protocol: WallpaperInteractor.self) as! WallpaperInteractor
    
    return WallpapersViewModel(interactor: interactor)
}

func getWallpaperViewModel(item: WallpaperIdentifiable) -> WallpaperViewModel {
    
    let interactor = koin.get(protocol: WallpaperInteractor.self) as! WallpaperInteractor
    
    let viewModel = WallpaperViewModel(interactor: interactor, data: item)
    
    return viewModel
}

func getCalendarViewModel() -> CalendarViewModel {
    
    let interactor = koin.get(protocol: WallpaperInteractor.self) as! WallpaperInteractor
    
    return CalendarViewModel(interactor: interactor)
}

func getBookmarksViewModel() -> BookmarksViewModel {
    
    let interactor = koin.get(protocol: WallpaperInteractor.self) as! WallpaperInteractor
    
    return BookmarksViewModel(interactor: interactor)
}

func getMoreAppsViewModel() -> MoreAppsViewModel {
    return MoreAppsViewModel()
}

func getLicensesViewModel() -> LicensesViewModel {
    
    let interactor = koin.get(protocol: LicenseInteractor.self) as! LicenseInteractor
    
    return LicensesViewModel(interactor: interactor)
}
